streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -256,40 +246,49 @@ public StateStore getStore(final String name) {
 
     @Override
     public void flush() {
+        ProcessorStateException firstException = null;
+        // attempting to flush the stores
         if (!stores.isEmpty()) {
             log.debug("Flushing all stores registered in the state manager");
             for (final StateStore store : stores.values()) {
+                log.trace("Flushing store {}", store.name());
                 try {
-                    log.trace("Flushing store={}", store.name());
                     store.flush();
                 } catch (final Exception e) {
-                    throw new ProcessorStateException(String.format("%sFailed to flush state store %s", logPrefix, store.name()), e);
+                    if (firstException == null) {
+                        firstException = new ProcessorStateException(String.format("%sFailed to flush state store %s", logPrefix, store.name()), e);
```

Comment:
Could e be ProcessorStateException ?
If yes, the new ProcessorStateException instance doesn't have to be created.

Reply:
That's actually possible in the current code base. Good catch.

Reply:
Yeah, i think it is better to add the additional context